source/common/stats/allocator_impl.cc

```diff
@@ -226,6 +223,35 @@ class GaugeImpl : public StatsSharedImpl<Gauge> {
       break;
     }
   }
+
+private:
+  std::atomic<uint64_t> value_{0};
+};
+
+class TextReadoutImpl : public StatsSharedImpl<TextReadout> {
+public:
+  TextReadoutImpl(StatName name, AllocatorImpl& alloc, StatName tag_extracted_name,
+                  const StatNameTagVector& stat_name_tags)
+      : StatsSharedImpl(name, alloc, tag_extracted_name, stat_name_tags) {}
+
+  void removeFromSetLockHeld() EXCLUSIVE_LOCKS_REQUIRED(alloc_.mutex_) override {
+    const size_t count = alloc_.text_readouts_.erase(statName());
+    ASSERT(count == 1);
+  }
+
+  // Stats::TextReadout
+  void set(const std::string& value) override {
+    Thread::LockGuard lock(mutex_);
+    value_ = value;
+  }
+  std::string value() const override {
+    Thread::LockGuard lock(mutex_);
+    return value_;
+  }
+
+private:
+  mutable Thread::MutexBasicLockable mutex_;
```

Comment:
can you make this an absl::Mutex directly? No need for this particular mutex to implement a pure interface, to be compatible with shared-memory. I think that will save some bytes per TextReadout.